statick_tool/plugins/tool/shellcheck_tool_plugin.py

```diff
@@ -0,0 +1,108 @@
+"""Apply shellcheck tool and gather results."""
+import argparse
+import json
+import subprocess
+from typing import Any, List, Optional
+
+from statick_tool.issue import Issue
+from statick_tool.package import Package
+from statick_tool.tool_plugin import ToolPlugin
+
+
+class ShellcheckToolPlugin(ToolPlugin):
+    """Apply shellcheck tool and gather results."""
+
+    def get_name(self) -> str:
+        """Get name of tool."""
+        return "shellcheck"
+
+    def gather_args(self, args: argparse.Namespace) -> None:
+        """Gather arguments."""
+        args.add_argument(
+            "--shellcheck-bin",
+            dest="shellcheck_bin",
+            type=str,
+            help="shellcheck binary path",
+        )
+
+    def scan(self, package: Package, level: str) -> Optional[List[Issue]]:
+        """Run tool and gather output."""
+        if "shell_src" not in package or not package["shell_src"]:
+            return []
+
+        shellcheck_bin = "shellcheck"  # type: str
+        if self.plugin_context and self.plugin_context.args.shellcheck_bin is not None:
+            shellcheck_bin = self.plugin_context.args.shellcheck_bin
+
+        # Get output in GCC format.
+        flags = ["-f", "json"]  # type: List[str]
```

Comment:
I think it would be good to add to the documentation that this plugin sets the output of shellcheck to json style to facilitate parsing.